vcd/resource_vcd_vapp_vm.go

```diff
@@ -616,8 +732,25 @@ func resourceVcdVAppVmRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.Set("name", vm.VM.Name)
-	if len(vm.VM.NetworkConnectionSection.NetworkConnection) > 0 {
+	networks := d.Get("networks").([]interface{})
+	network := d.Get("network_name").(string)
+	switch {
+	// network_name is not set. networks is set in config
+	case network != "":
 		d.Set("ip", vm.VM.NetworkConnectionSection.NetworkConnection[0].IPAddress)
+		d.Set("mac", vm.VM.NetworkConnectionSection.NetworkConnection[0].MACAddress)
+	case len(networks) > 0:
+		var nets []map[string]interface{}
+		for index, net := range d.Get("networks").([]interface{}) {
+			n := net.(map[string]interface{})
+			if len(vm.VM.NetworkConnectionSection.NetworkConnection) > 0 {
```

Comment:
Imagine I've got 1 NIC in VM and 2 NICs in .tf. It would still hit an `index out of range` panic.
I think in `case len(networks) > 0:` (which means we're using the newer functionality) it is worth to loop over `vm.VM.NetworkConnectionSection.NetworkConnection` rather than `d.Get("networks").([]interface{})`. That way you will save the real state in vCD into the `terraform.tfstate` file.
What are your thoughts on doing it this way round?

Reply:
I do understand your concern and I'll change it.